File: Backend/src/main/java/com/project/consphere/service/FilterService.java
package com.project.consphere.service;

import com.project.consphere.dto.CreateFilterRequest;
import com.project.consphere.dto.FilterResponse;
import com.project.consphere.dto.PostResponse;
import com.project.consphere.dto.UpdateFilterRequest;

import java.util.List;

public interface FilterService {

    FilterResponse createFilter(CreateFilterRequest request);

    List<FilterResponse> getAllFilters();

    List<FilterResponse> getMyFilters();

    List<FilterResponse> getBuiltInFilters();

    FilterResponse getFilterById(Long id);

    FilterResponse updateFilter(Long id, UpdateFilterRequest request);

    void deleteFilter(Long id);

    List<PostResponse> getFilterFeed(Long id);
}

File: frontend/src/services/filterService.js
import api from './api';

export const filterService = {
  getAllFilters: async () => {
    const response = await api.get('/filters');
    return response.data;
  },

  getMyFilters: async () => {
    const response = await api.get('/filters/my-filters');
    return response.data;
  },

  getBuiltInFilters: async () => {
    const response = await api.get('/filters/built-in');
    return response.data;
  },

  getFilterById: async (id) => {
    const response = await api.get(`/filters/${id}`);
    return response.data;
  },

  createFilter: async (filterData) => {
    const response = await api.post('/filters', filterData);
    return response.data;
  },

  updateFilter: async (id, filterData) => {
    const response = await api.put(`/filters/${id}`, filterData);
    return response.data;
  },

  deleteFilter: async (id) => {
    const response = await api.delete(`/filters/${id}`);
    return response.data;
  },

  getFilterFeed: async (id) => {
    const response = await api.get(`/filters/${id}/feed`);
    return response.data;
  }
};
